File: src/main/java/com/autosale/shop/model/Product.java
package com.autosale.shop.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Value;

@Value
public class Product {

    Integer id;
    String name;
    String description;
    Float cost;
    ProductStatus status;
    Integer sellerId;
    Integer buyerId;

    public Product(@JsonProperty("id") Integer id, @JsonProperty("name") String name, @JsonProperty("description") String description, @JsonProperty("cost") Float cost, @JsonProperty("status") ProductStatus status, @JsonProperty("seller_id") Integer sellerId, @JsonProperty("buyer_id") Integer buyerId) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.cost = cost;
        this.status = status;
        this.sellerId = sellerId;
        this.buyerId = buyerId;
    }
}

File: src/main/java/com/autosale/shop/model/Product.kt
package com.autosale.shop.model

import com.fasterxml.jackson.annotation.JsonProperty

data class Product(
    @JsonProperty("id") val id: Int,
    @JsonProperty("name") val name: String,
    @JsonProperty("description") val description: String,
    @JsonProperty("cost") val cost: Float,
    @JsonProperty("status") val status: ProductStatus?,
    @JsonProperty("seller_id") val sellerId: Int?,
    @JsonProperty("buyer_id") val buyerId: Int?
)
